src/app/modules/project/add/manual/manual.component.ts

```diff
@@ -108,4 +121,16 @@ export class ManualComponent implements OnInit {
     }
     this.collaborators.splice(index, 1);
   }
+
+  /**
+   * Method to fill a form with the values of a mapped project.
+   */
+  private fillFormWithProject(project: MappedProject): void {
+    this.newProjectForm.get('name').setValue(project.name);
+    this.newProjectForm.get('uri').setValue(project.uri);
+    this.newProjectForm.get('shortDescription').setValue(project.shortDescription);
+    this.newProjectForm.get('description').setValue(project.description);
+
```

Comment:
What if one of those values from the MappedProject is empty? Would it clear previous input in the form?

Reply:
The fill method is excuted as the fetchedSource is received in the ngOnInit. Meaning that the form will be filled before a user can even input any value. If the values is null the input will remain empty and the user will need to fill in the values themself.